.github/workflows/deploy-to-eks.yaml

```
@@ -0,0 +1,205 @@
+name: Create EKS cluster, deploy kubeflow-mlflow Terraform solution and run UATs
+
+on:
+  pull_request: {}
+  workflow_dispatch:
+    inputs:
+      k8s_version:
+        description: 'Kubernetes version to use for the EKS cluster (e.g. 1.27)'
+        required: false
+      uats_branch:
+        description: 'Branch to run the UATs from, e.g., main or track/1.10'
+        required: false
+  schedule:
+    - cron: "17 02 * * 1"
+
+env:
+  CLUSTER_NAME: kubeflow-eks2
+
+jobs:
+  deploy-solution-to-eks:
+    name: Deploy CKF + MLFlow solution to EKS
+    runs-on: ubuntu-24.04
+    outputs:
+      aws_region: ${{ steps.extract_region.outputs.region }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set envvars from dependencies.yaml
+        run: |
+          yq eval 'to_entries | .[] | "\(.key)=\(.value)"' ".github/dependencies.yaml" | while IFS= read -r line; do
+            echo "$line" >> "$GITHUB_ENV"
+          done
+
+      - name: Update ENV variables from inputs if available
+        run: |
+          K8S_VERSION=${{ inputs.k8s_version || env.K8S_VERSION }}
+          echo "K8S_VERSION=${K8S_VERSION}" >> $GITHUB_ENV
+          UATS_BRANCH=${{ inputs.uats_branch || env.UATS_BRANCH }}
+          echo "UATS_BRANCH=${UATS_BRANCH}" >> $GITHUB_ENV
+
+      - name: Extract AWS region from cluster.yaml
+        id: extract_region
+        run: |
+          REGION=$(yq e '.metadata.region' .github/cluster.yaml)
+          echo "AWS_REGION=$REGION" >> $GITHUB_ENV
+          echo "region=$REGION" >> $GITHUB_OUTPUT
+
+      # Double-check if this is still needed - same with pinning Charmcraft version
+      - name: Set up Python 3.12
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+
+      - name: Install CLI tools & dependencies
+        run: |
+          pip install tox
+          sudo snap install juju --channel=${{ env.JUJU_VERSION }}/stable
+          sudo snap install charmcraft --classic
+          sudo snap refresh charmcraft --channel=3.x/stable
+          sudo snap install terraform --channel=latest/stable --classic
```

nitpick: One-liner suggestion:
```suggestion
          sudo snap refresh charmcraft --channel=3.x/stable
          sudo snap install terraform --channel=latest/stable --classic
```